lib/home/surveys/take_survey/survey.dart

```diff
@@ -41,19 +41,27 @@ class _SurveyScreenState extends State<SurveyScreen> {
     if (FunQuiz.inProgress) {
       navigator.pushReplacement(FunQuizResults(data.funQuizResults));
       return;
-    } else if (data.valid) {
+    }
+    final validation = context.read<ValidSurveyAnswers>();
+    if (data.valid) {
+      validation.emit(false);
       navigator.pushReplacement(Submitted(data.summary));
       return;
     }
-    context.read<SurveyValidation>().submit();
+    validation.emit(true);
   }
 
   /// Creates a function for each [SurveyField] that can update the survey [data].
   ValueChanged<dynamic> update(int i, SurveyRecord record) {
     (dynamic, String?) fromMultipleChoice((dynamic, String?) newAnswer) {
       final (oldData, String? oldInput) = record.answer ?? (null, null);
       final (newData, String? newInput) = newAnswer;
-      return (newData ?? oldData, newInput.validated ?? oldInput);
+      final answer = switch (newInput) {
+        String() when newInput.valid => newInput,
+        String() => null, // blank answer
+        null => oldInput,
+      };
+      return (newData ?? oldData, answer);
```

Comment:
Small bugfix here: It used to be that if a user typed out a response and then deleted it, it would be saved as just the first letter, which is probably not what we would want :D